cpg_pipes/providers/status.py

```diff
@@ -0,0 +1,60 @@
+"""
+Abstract reporter of stages statuses.
+"""
+
+import logging
+from abc import ABC, abstractmethod
+from enum import Enum
+
+from hailtop.batch.job import Job
+from hailtop.batch import Batch
+
+from .. import Path
+from ..targets import Target
+
+logger = logging.getLogger(__file__)
+
+
+class StatusReporterError(Exception):
+    """
+    Raised for problems reporting status
+    """
+
+
+class AnalysisStatus(Enum):
```

Comment:
I'm a little concerned about this reimplementation, why not just pull it from sample-metadata and subclass it - then if we add / remove statuses it will fail before it even starts.

Reply:
The idea was to have the abstract `StatusReporter` independent of sample-metadata implementation, thus it can't import sample-metadata. Not sure if there is a better way to work around it.